### PR TITLE
feat(tab): add trailing icon option (VIV-1109)

### PR DESCRIPTION
**Pull request by rachelbt**

### CODE REVIEW

File: libs/components/src/lib/tab/tab.spec.ts [hunk 1]
```diff
@@ -30,6 +34,29 @@ describe('vwc-tab', () => {
 		});
 	});
 
+	describe('tab icon', () => {
```
Comment (YonatanKra) [inline libs/components/src/lib/tab/tab.spec.ts:37]:
```suggestion
	describe('icon', () => {
```

File: libs/components/src/lib/tab/tab.spec.ts [hunk 2]
```diff
@@ -30,6 +34,29 @@ describe('vwc-tab', () => {
...
+		it('setting `iconTrailing` set the order of element', async () => {
+			element.icon = 'home';
+			element.iconTrailing = true;
+			await elementUpdated(element);
+
+			const trailingIcon = element.shadowRoot?.querySelector(
+				`.icon-trailing ${ICON_SELECTOR}`,
+			);
+			expect(trailingIcon).toBeInstanceOf(HTMLElement);
+		});
```
Comment (YonatanKra) [inline libs/components/src/lib/tab/tab.spec.ts:56]: Extract it to its own `describe` as this is an API unto itself.
Also, the description is vague and I can't see how the test tests it.
What should it do?
It looks as if the code just adds the class. Am I correct?

File: libs/components/src/lib/tab/tab.spec.ts [hunk 3]
```diff
@@ -30,6 +34,29 @@ describe('vwc-tab', () => {
 		});
 	});
 
+	describe('tab icon', () => {
+		it('adds an icon to the tab', async () => {
```
Comment (YonatanKra) [inline libs/components/src/lib/tab/tab.spec.ts:38]:
```suggestion
		it('should add an icon to the tab when icon is set', async () => {
```

**Review by YonatanKra — CHANGES_REQUESTED** [review]

**Review by olaf-k — DISMISSED** [review]

File: libs/components/src/lib/tab/tab.spec.ts (hunk 2 above)
Comment (rachelbt) [reply libs/components/src/lib/tab/tab.spec.ts]: Also - in all out unittest the trailing icon is under the icon describe

**Review by YonatanKra — APPROVED** [review]
LGTM

**Pull request merged**